### PR TITLE
Blac 44 check and apply for open vacancy

### PR DESCRIPTION
**Pull request by kimkaminska**

### CODE REVIEW

**Review by VityaPodill — APPROVED** [review]

**Review by Bogdan27nazaruk — APPROVED** [review]

**Review by Bogdan27nazaruk — APPROVED** [review]

File: src/test/java/elements/ApplyForOpenJobElements.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.openqa.selenium.WebElement;
+
+public class ApplyForOpenJobElements extends AbstractBasePage {
+
+    public ApplyForOpenJobElements(WebDriver driver) {
+        super(driver);
+    }
+
+    private static final String APPLY_NOW_BUTTON = "//a[@class='btn btn-cta']";
+    private static final String ENTER_USERS_EMAIL = "//input[@data-automation-id='email']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/ApplyForOpenJobElements.java]: please create a method dataAutomationId() the same as dataTestId() and use it here

File: src/test/java/elements/ApplyForOpenJobElements.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private static final String ENTER_USERS_PASSWORD = "//input[@data-automation-id='password']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/ApplyForOpenJobElements.java]: ditto

File: src/test/java/elements/CareersElements.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+public class CareersElements extends AbstractBasePage {
+
+    public CareersElements(WebDriver driver) {
+        super(driver);
+    }
+
+    private static final String EXPLORE_NOW_CORPORATE_JOBS = "//a[text() = 'EXPLORE NOW '][1]";
+    private static final String CORPORATE_PREFILTER = "//div[contains(text(), 'Corporate')]";
+    private static final String ACCEPT_ALL_COOKIES = "//button[@id='onetrust-accept-btn-handler']";
+    private static final String LOCATIONS_FILTER = "//div[@class= 'jobfinder-filter-list-container job-corporate active']//span[text()= 'Locations']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/CareersElements.java]: there are two unnecessary spaces it this locator

File: src/test/java/elements/CareersElements.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    private static final String FUTURE_TALENTS_FILTER = "//div[@class= 'jobfinder-filter-list-container job-corporate active']//span[text()= 'Future Talents']";
+    private static final String SECTION_OF_FUTURE_TALENTS_FILTER = "//div[@class='jobfinder-filter-list last-level']//span[text()= '%s']";
+    private static final String SELECTED_OPEN_JOBS_LIST_WITH_REGARD_TO_FUTURE_TALENTS_FILTER = "//p[@class='team']";
+    private static final String NEEDED_OPEN_JOB = "(//div[@class= 'content']//h3)[1]";
+    private static final String SIGN_IN_BUTTON = "//div[@aria-label='Sign In']";
+    private static final String NO_RESULT = "//p[@class='no-result']";
+
+    public WebElement getExploreNowCorporateJobs() {
+        return waitUntilElementToBeVisibleByXpath(EXPLORE_NOW_CORPORATE_JOBS);
+    }
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/CareersElements.java]: your style of getters are different from other developers in this framework. Please choose one style

**Review by Bogdan27nazaruk — APPROVED** [review]

File: src/test/java/elements/HomeElements.java [hunk 5]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
         super(driver);
     }
 
+    private static final String CLICK_HERE_TO_EXPLORE_CAREERS = "//section//a[text() = 'CLICK HERE']";
     public static final String CATEGORY = "//a[@data-link-name='%s']";
     public static final String WOMEN_CLASSICS_CATEGORY = "//ul[@aria-label='Shoes']//a[@data-link-name='Classics']";
     public static final String SELECT_LOCATION_CLOSE_BTN = "//button[@data-test-id='close-btn']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: use dataTestId() method please

File: src/test/java/elements/HomeElements.java [hunk 6]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
     public static final String COOKIE_CLOSE_BTN = "//button[@data-test-id='cookie-banner-close-btn']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: ditto

File: src/test/java/elements/HomeElements.java [hunk 7]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
     protected static final String POP_UP_MENU_SALE_WOMEN_ACCESSORIES = "//a[@href='/us/en/sale/womens-sale/accessories']";
-
     //homepage elements
+
     public static final String RETURN_HOME_PAGE_PUMA_ICON = "//a[@data-test-id='main-nav-home-link']";
     public static final String ACCOUNT_ICON_BUTTON = "//button[@data-test-id='account-button']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: why are these variables public?

File: src/test/java/elements/HomeElements.java [hunk 8]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
-
     //account menu elements
+
     public static final String LOGIN_BUTTON_ON_ACCOUNT_MENU = "//a[@data-test-id='login-button']";
     public static final String REGISTER_HERE_BUTTON = "//a[@data-test-id='register-button']";
     public static final String COLLABORATION_BTN = "//span[contains(text(),'Collaborations')]";
     public static final String SEARCH_BTN ="//button[@type='submit']//div";
     public static final String SEARCH_INPUT_FIELD = "//input[@placeholder='SEARCH PUMA.COM']";
     public static final String SEARCH_FIELD = "//button[@data-test-id='search-icon-nav']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: public?

File: src/test/java/elements/HomeElements.java [hunk 9]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
 
+    private static final String CLICK_HERE_TO_EXPLORE_CAREERS = "//section//a[text() = 'CLICK HERE']";
     public static final String CATEGORY = "//a[@data-link-name='%s']";
     public static final String WOMEN_CLASSICS_CATEGORY = "//ul[@aria-label='Shoes']//a[@data-link-name='Classics']";
     public static final String SELECT_LOCATION_CLOSE_BTN = "//button[@data-test-id='close-btn']";
     public static final String COOKIE_CLOSE_BTN = "//button[@data-test-id='cookie-banner-close-btn']";
     protected static final String POP_UP_MENU_SALE_WOMEN_ACCESSORIES = "//a[@href='/us/en/sale/womens-sale/accessories']";
-
     //homepage elements
+
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java:21]: remove empty line

File: src/test/java/elements/HomeElements.java [hunk 10]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
     public static final String RETURN_HOME_PAGE_PUMA_ICON = "//a[@data-test-id='main-nav-home-link']";
     public static final String ACCOUNT_ICON_BUTTON = "//button[@data-test-id='account-button']";
-
     //account menu elements
+
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java:25]: remove empty line

File: src/test/java/elements/HomeElements.java [hunk 11]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
     public static final String LOGIN_BUTTON_ON_ACCOUNT_MENU = "//a[@data-test-id='login-button']";
     public static final String REGISTER_HERE_BUTTON = "//a[@data-test-id='register-button']";
     public static final String COLLABORATION_BTN = "//span[contains(text(),'Collaborations')]";
     public static final String SEARCH_BTN ="//button[@type='submit']//div";
     public static final String SEARCH_INPUT_FIELD = "//input[@placeholder='SEARCH PUMA.COM']";
     public static final String SEARCH_FIELD = "//button[@data-test-id='search-icon-nav']";
-
     //men menu elements
+
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: remove empty line

File: src/test/java/elements/HomeElements.java [hunk 12]
```diff
@@ -12,25 +12,26 @@ public HomeElements(WebDriver driver) {
...
     public static final String SOCCER_SHOES_MEN_CATEGORY = "(//a[@data-link-name='Soccer'])[1]";
     public static final String BASKETBALL_SHOES_MEN_CATEGORY = "//a[@data-link-name='Basketball']";
     public static final String HATS_ACCESSORIES_MEN_CATEGORY = "//a[@data-link-name='Hats']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: public?

File: src/test/java/pages/CareersPage.java [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        return this;
+    }
+
+    public CareersPage clickAcceptAllCookies() {
+        getAcceptAllCookies().click();
+        return this;
+    }
+
+    public CareersPage clickClearFilters() throws InterruptedException {
+        sleep(3000);
```
Comment (AndriiMyronenko91) [inline src/test/java/pages/CareersPage.java:52]: why do you need sleep here?

File: src/test/java/pages/CareersPage.java [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+        return this;
+    }
+
+    public CareersPage clickBackToPreviousStepSelectingLocation() {
+        getBackToPreviousStepSelectingLocationButton().click();
+        return this;
+    }
+
+    public CareersPage clickTeamsFilter() throws InterruptedException {
+        sleep(3000);
```
Comment (AndriiMyronenko91) [inline src/test/java/pages/CareersPage.java:63]: sleep?

File: src/test/java/pages/CareersPage.java [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+        return this;
+    }
+
+    public CareersPage clickFutureTalentsFilter() {
+        getFutureTalentsFilter().click();
+        return this;
+    }
+
+    public CareersPage clickSectionOfFutureTalentsFilter(String option) throws InterruptedException {
+        sleep(2000);
```
Comment (AndriiMyronenko91) [inline src/test/java/pages/CareersPage.java:79]: sleep?

File: src/test/java/pages/CareersPage.java [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+        getSectionOfFutureTalentsFilter(option).click();
+        return this;
+    }
+
+    public CareersPage clickOnNeededOpenJob() throws InterruptedException {
+        sleep(3000);
```
Comment (AndriiMyronenko91) [inline src/test/java/pages/CareersPage.java:85]: sleep?

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+        homePage
+                .clickGetClickHereToExploreCareers();
+
+        switchToTab(2);
+
+        softAssert.assertTrue(getCurrentUrl().contains("/careers"));
+
+        careersPage
+                .clickExploreNowCorporateJobs()
+                //test fail at this step from time to time
```
Comment (AndriiMyronenko91) [inline src/test/java/tests/CheckAndApplyForOpenVacancy.java]: stabilize it and remove comment please

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+
+        for (int j = 0; j < careersPage.getSelectedOpenJobsList().size(); j++) {
+            softAssert.assertTrue(careersPage.getSelectedOpenJobsCityOfLocation().get(j).getText().contains("Salzburg"), "Location filter doesn't work correctly.");
+        }
+
+        careersPage
+                .clickBackToPreviousStepSelectingLocation()
+                .clickBackToPreviousStepSelectingLocation()
+                .clickBackToPreviousStepSelectingLocation()
+                //test fail at this step from time to time
```
Comment (AndriiMyronenko91) [inline src/test/java/tests/CheckAndApplyForOpenVacancy.java]: ditto

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+                .clickSectionOfFutureTalentsFilter("Apprentices");
+
+        for (int j = 0; j < careersPage.getSelectedOpenJobsList().size(); j++) {
+            softAssert.assertTrue(careersPage.getSelectedOpenJobsListWithRegardToFutureTalentsFilter().get(j).getText().contains("Apprentices"), "Future Talents filter doesn't work correctly.");
+        }
+
+        careersPage
+                .clickClearFilters()
+                .clickSectionOfFutureTalentsFilter("Interns")
+                //test fail at this step from time to time
```
Comment (AndriiMyronenko91) [inline src/test/java/tests/CheckAndApplyForOpenVacancy.java]: ditto

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+                .clickOnNeededOpenJob();
+
+        applyForOpenJobPage
+                //test fail at this step from time to time
```
Comment (AndriiMyronenko91) [inline src/test/java/tests/CheckAndApplyForOpenVacancy.java]: ditto

**Review by NataLitovets — APPROVED** [review]

**Review by AndriiMyronenko91 — CHANGES_REQUESTED** [review]

File: src/test/java/elements/ApplyForOpenJobElements.java (hunk 1 above)
Comment (kimkaminska) [reply src/test/java/elements/ApplyForOpenJobElements.java]: Ok, thanks

Comment (kimkaminska) [reply src/test/java/elements/ApplyForOpenJobElements.java]: done

File: src/test/java/elements/ApplyForOpenJobElements.java (hunk 2 above)
Comment (kimkaminska) [reply src/test/java/elements/ApplyForOpenJobElements.java]: Thanks. Done.

File: src/test/java/elements/CareersElements.java (hunk 3 above)
Comment (kimkaminska) [reply src/test/java/elements/CareersElements.java]: Thanks, corrected

File: src/test/java/elements/CareersElements.java (hunk 4 above)
Comment (kimkaminska) [reply src/test/java/elements/CareersElements.java]: Thanks. We've already discussed with You this comment.

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java (hunk 17 above)
Comment (kimkaminska) [reply src/test/java/tests/CheckAndApplyForOpenVacancy.java]: Ok, thanks

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java (hunk 18 above)
Comment (kimkaminska) [reply src/test/java/tests/CheckAndApplyForOpenVacancy.java]: Ok, thanks

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java (hunk 19 above)
Comment (kimkaminska) [reply src/test/java/tests/CheckAndApplyForOpenVacancy.java]: Ok, thanks

File: src/test/java/tests/CheckAndApplyForOpenVacancy.java (hunk 20 above)
Comment (kimkaminska) [reply src/test/java/tests/CheckAndApplyForOpenVacancy.java]: Ok, thanks

File: src/test/java/elements/HomeElements.java (hunk 5 above)
Comment (kimkaminska) [reply src/test/java/elements/HomeElements.java]: Thanks, Ok

File: src/test/java/pages/CareersPage.java (hunk 16 above)
Comment (kimkaminska) [reply src/test/java/pages/CareersPage.java:85]: Test failed without sleep. I'll try to avoid sleep.

**Pull request merged**